tests/serverpod_test_server/test_e2e_migrations/migrations_roundtrip_test.dart

```diff
@@ -0,0 +1,995 @@
+@Timeout(Duration(minutes: 5))
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:serverpod_service_client/serverpod_service_client.dart';
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:serverpod_test_server/test_util/service_key_manager.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var serviceClient = Client(
+    serviceServerUrl,
+    authenticationKeyManager: ServiceKeyManager('0', 'password'),
+  );
+
+  group('Given new protocol entity with table', () {
+    var scenario = 'add-table';
+    tearDown(() async {
+      await _migrationTestCleanup(
+        scenario: scenario,
+        serviceClient: serviceClient,
+      );
+    });
+
+    test(
+        'when creating and applying migration then database contains new table.',
+        () async {
+      _copyTestProtocolFiles(
+        scenario: scenario,
+        assetType: AssetType.targetState,
+      );
+
+      var createMigrationExitCode = await _runCreateMigrations(scenario);
+      expect(
+        await createMigrationExitCode,
+        0,
+        reason: 'Failed to create migration, exit code was not 0.',
+      );
+
+      var applyMigrationExitCode = await _runApplyMigrations();
+      expect(
+        await applyMigrationExitCode,
+        0,
+        reason: 'Failed to apply migration, exit code was not 0.',
+      );
+
+      var addedTableName = 'migrated_table';
+      var liveDefinition =
+          await serviceClient.insights.getLiveDatabaseDefinition();
+      var databaseTables = liveDefinition.tables.map((t) => t.name);
+      expect(
+        databaseTables,
+        contains(addedTableName),
+        reason: 'Could not find migration table in live table definitions.',
+      );
+    });
+  });
+
+  group('Given multiple new protocol entities with table', () {
+    var scenario = 'add-multiple-tables';
+    tearDown(() async {
+      await _migrationTestCleanup(
+        scenario: scenario,
+        serviceClient: serviceClient,
+      );
+    });
+
```

Comment:
Suggestion: When we have a single test like here I think we could move the clean up into the test instead. If we do that we no longer need the wrapping group and we can leave the test flat in the main method. 

Less code, and I think it becomes clearer as well!

Reply:
We need the cleanup method since we want to clear the database even if the test fails.